apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/RestResourceBase.java

```diff
@@ -80,6 +128,26 @@ protected QueryOuterClass.Response queryWithTable(
         });
   }
 
+  protected Uni<QueryOuterClass.Response> executeQueryAsync(QueryOuterClass.Query query) {
+    return requestInfo.getStargateBridge().executeQuery(query);
+  }
+
+  protected Uni<QueryOuterClass.Response> executeQueryAsync(
+      String keyspaceName,
+      String tableName,
+      Function<Optional<Schema.CqlTable>, QueryOuterClass.Query> queryProducer) {
+
+    // TODO implement optimistic queries (probably requires changes directly in SchemaManager)
```

Comment:
What about this, should be very simple imo? And I don't think there is anything do to with the Schema Manager:

```
return findTableAsync(keyspaceName, tableName, true)
    .onItem()
    .transformToUni(table -> executeQueryAsync(queryProducer.apply(table)))
    .onFailure(t -> {// predicate for matching the optimistic schema failure}).retry().atMost(2);


```

Reply:
> What about this, should be very simple imo? And I don't think there is anything do to with the Schema Manager:
> 
> ```
> return findTableAsync(keyspaceName, tableName, true)
>     .onItem()
>     .transformToUni(table -> executeQueryAsync(queryProducer.apply(table)))
>     .onFailure(t -> {// predicate for matching the optimistic schema failure}).retry().atMost(2);
> ```

It was an existing TODO, did not look into it yet. But sounds reasonable, adding another issue :)